src/combinatorics/set_family/mathlib.lean

```diff
@@ -0,0 +1,67 @@
+/-
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import order.upper_lower.basic
+import tactic.by_contra
+
+section linear_order
+variables {α : Type*} [linear_order α] {s t : set α}
+
+lemma is_lower_set.total (hs : is_lower_set s) (ht : is_lower_set t) : s ⊆ t ∨ t ⊆ s :=
+begin
+  by_contra' h,
+  simp_rw set.not_subset at h,
+  obtain ⟨⟨a, has, hat⟩, b, hbt, hbs⟩ := h,
+  obtain hab | hba := le_total a b,
+  { exact hat (ht hab hbt) },
+  { exact hbs (hs hba has) }
+end
+
+lemma is_upper_set.total (hs : is_upper_set s) (ht : is_upper_set t) : s ⊆ t ∨ t ⊆ s :=
+begin
+  by_contra' h,
+  simp_rw set.not_subset at h,
+  obtain ⟨⟨a, has, hat⟩, b, hbt, hbs⟩ := h,
+  obtain hab | hba := le_total a b,
+  { exact hbs (hs hab has) },
+  { exact hat (ht hba hbt) }
+end
```

Comment:
Does this come for free via `order_dual`?

Reply:
I thought not but it's just because I accidentally swapped the names of `is_upper_set.to_dual`/`is_upper_set.of_dual`.